src/registries/npm-registry.js

```diff
@@ -218,21 +221,27 @@ export default class NpmRegistry extends Registry {
     }
   }
 
-  getScope(packageName: string): string {
-    return !packageName || packageName[0] !== '@' ? '' : packageName.split(/\/|%2f/)[0];
+  getScope(packageIdent: string): string {
+    // Matches '@scope' in a packageName or pathname, otherwise returns ''
+    if (packageIdent && isScopedPackage(packageIdent)) {
+      const matched = packageIdent.match(/(^@|(?!\/)@).+?((?=%2f)|(?=\/))/);
```

Comment:
This regular expression is very hard to follow, can use some comments. Also what's with the `%2f` part? We should unescape the URL and run the matcher on it instead of doing this.